File: ch1/ch1-pg11-exam4.cpp
#include<stdio.h>

int main() {

    int numOfHeads,numOfLegs;
    int numOfChickens, numOfRabbits;

    scanf("%d\t%d", &numOfHeads, &numOfLegs);

    if (numOfHeads <= 0 || numOfLegs <= 0 || numOfLegs % 2 == 1) {
        printf("No answer\n");
        return 0;
    }
 
    numOfChickens = (numOfHeads * 4 - numOfLegs) / 2;
    numOfRabbits = numOfHeads - numOfChickens;

    if (numOfChickens < 0 || numOfRabbits < 0) {
        printf("No answer\n");
    } else {
        printf("%d  %d", numOfChickens, numOfRabbits);
    }

    return 0;
}

File: ch1/ch1-pg11-exam4.js
const lnrd = require("readline");
const consolLnrd = lnrd.createInterface({
    input: process.stdin
});
consolLnrd.on("close", () => {
    process.exit()
});

let numOfHeads = 0;
let numOfLegs = 0;
let numOfChicken = -1;
let numOfRabits = -1;

let counter = 1;

consolLnrd.on("line", input => {
    if (counter == 1) {
        numOfHeads = input;
        if (numOfHeads % 2 != 0) {
            console.log("The number of heads should be divided by 2");
            counter--;
        }
    }
    if (counter == 2) {
        numOfLegs = input;
        if (numOfLegs % 2 != 0) {
            console.log("The number of legs should be divided by 2");
            counter--;
        } else {
            console.log("The number of heads is:", `${numOfHeads}`);
            console.log("The number of legs is:", `${numOfLegs}`);
            numOfRabits = numOfLegs / 2 - numOfHeads;
            numOfChicken = numOfHeads - numOfRabits;

            if (numOfRabits < 0 || numOfChicken < 0) {
                console.log("There is no answer according to the input number of heads and legs");
                consolLnrd.close();
            }

            console.log("The number of chicken is:", `${numOfChicken}`);
            console.log("The number of rabits is:", `${numOfRabits}`);
            consolLnrd.close();
        }
    }
    counter++;
});
